src/repositories/books/books.repository.ts

```diff
@@ -1,19 +1,67 @@
-import { readFile } from '../../helpers/helpers.ts';
-import { Book } from '../../common/types/types.ts';
-import { IRepository } from '../../common/interfaces/interfaces.ts';
-
-const booksDataPath = new URL('./books.json', import.meta.url).pathname;
-
-class Books implements Partial<IRepository<Book>> {
-  public findAll(): Promise<Book[]> {
-    return this._getBooks();
-  }
-
-  private async _getBooks(): Promise<Book[]> {
-    const books = await readFile(booksDataPath);
-
-    return JSON.parse(books);
-  }
-}
-
-export { Books };
+import { readFile, writeFile } from '../../helpers/helpers.ts';
+import { Book } from '../../common/types/types.ts';
+import { IRepository } from '../../common/interfaces/interfaces.ts';
+
+const booksDataPath = new URL('./books.json', import.meta.url).pathname;
+
+class Books implements IRepository<Book> {
+  public findAll(): Promise<Book[]> {
+    return this._getBooks();
+  }
+
+  public async findOne(id: string): Promise<Book | null> {
+	const books = await this._getBooks();
+	const book = books.find(book => book.id === id);

```

Comment:
It would be better to concert all these things into function and move them to `src/repositories/books/helers/`